File: src/mappings/torus/mapc2m_twisted_torus.f
      subroutine mapc2m_twisted_torus(blockno,xc,yc,xp,yp,zp,alpha)
      implicit none

      integer blockno
      double precision xc,yc,xp,yp,zp
      double precision alpha, r

      double precision pi, pi2
      common /compi/ pi, pi2

      r = 1.d0 + alpha*cos(pi2*(yc + xc))

      xp = r*cos(pi2*xc)
      yp = r*sin(pi2*xc)
      zp = alpha*sin(pi2*(yc + xc))

      end

File: matlab/mapc2m_twisted_torus.m
function [xp,yp,zp] = mapc2m_twisted_torus(xc,yc,alpha)

% alpha = 0.4;  % outer radius/inner radius

r = 1 + alpha*cos(2*pi*(xc+yc));
zp = alpha.*sin(2*pi*(xc+yc));
tw = 0;
th = 2*pi*(xc - tw*zp/0.4);
xp = r.*cos(th);
yp = r.*sin(th);

end